packages/console/src/pages/Connectors/components/Guide/index.tsx

```diff
@@ -49,7 +49,7 @@ const Guide = ({ connector, onClose }: Props) => {
       return;
     }
 
-    const result = arbitraryObjectGuard.safeParse(connectorConfigJson);
+    const result = arbitraryObjectGuard.safeParse(JSON.parse(connectorConfigJson));
```

Comment:
JSON parse will throw

Reply:
then maybe we should implement a `safeParseJson` as we have many calls on `JSON.parse()`.

Reply:
Ah, yes...

Reply:
Then I would say the best way to handle this in this situation is keeping `as` and the `try-catch` block. Thought using zod would be easier but it doesn't work as I expected.

Reply:
Implemented a `safeParseJson` util. PTAL